src/components/settings/SettingsTextInput.jsx

```diff
@@ -21,6 +21,7 @@ export default function SettingsTextInput({
   setCurrentValues,
   customFieldCategories,
   settingKey,
+  skipDescription = false,
```

Comment:
instead of having a prop here, could you just check whether `matchingSetting.description || matchingSetting.descriptionId` exists?